sdk/identity/azure-identity/inc/azure/identity/managed_identity_credential.hpp

```diff
@@ -71,6 +71,22 @@ namespace Azure { namespace Identity {
         Azure::Core::Credentials::TokenCredentialOptions const& options
         = Azure::Core::Credentials::TokenCredentialOptions());
 
+    /**
+     * @brief Constructs an instance of ManagedIdentityCredential capable of authenticating a
+     * resource with a user-assigned managed identity.
+     *
+     * @remark Client ID and object ID are NOT interchangeable, even though they are both Uuid
+     * values. Make sure to the pass in the correct value to the corresponding constructor.
+     *
+     * @param objectId The object ID of the service principal used to authenticate a user-assigned
+     * managed identity. This is an alternative to providing a client ID, and isn't required for
+     * system-assigned managed identities.
+     * @param options Options for token retrieval.
```

Comment:
Since the documentation linked above doesn't include object ID or client ID user defined identity, it might be helpful to link to the documentation for those scenarios (I could only find documentation for a resource ID user defined identity in the linked article).

Reply:
I haven't found the canonical documentation to link to, yet, and I want to be cautious in which link we promote (to avoid linking to info that could get stale). Different endpoint/hosts have their REST endpoint documentation that mention specifics. I could add links to those, but they might cause confusion.

Let me see if others have some good reference material that they add to the doc comments.

I am hoping, a doc like this will help clarify the different IDs that can be used to map a user-assigned MI:
https://github.com/Azure/azure-sdk-for-cpp/pull/5838